src/main/java/eu/adlogix/appnexus/oas/client/service/CampaignService.java

```diff
@@ -0,0 +1,111 @@
+package eu.adlogix.appnexus.oas.client.service;
+
+import static eu.adlogix.appnexus.oas.client.utils.ValidatorUtils.checkNotEmpty;
+import static eu.adlogix.appnexus.oas.client.utils.ValidatorUtils.checkNotNull;
+import static org.apache.commons.lang3.StringUtils.defaultIfEmpty;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import eu.adlogix.appnexus.oas.client.domain.Campaign;
+import eu.adlogix.appnexus.oas.client.domain.StatefulDomainManager;
+import eu.adlogix.appnexus.oas.client.domain.enums.BillTo;
+import eu.adlogix.appnexus.oas.client.domain.enums.CampaignType;
+import eu.adlogix.appnexus.oas.client.parser.XmlToCampaignParser;
+import eu.adlogix.appnexus.oas.client.transform.CampaignCreateParameterMapTransformer;
+import eu.adlogix.appnexus.oas.client.transform.CampaignParameterMapTransformer;
+import eu.adlogix.appnexus.oas.client.transform.CampaignUpdateParameterMapTransformer;
+import eu.adlogix.appnexus.oas.client.xml.ResponseParser;
+import eu.adlogix.appnexus.oas.client.xml.XmlRequestGenerator;
+
+/**
+ * Service Class which provides functions for all Campaign related operations
+ * 
+ */
+public class CampaignService extends AbstractOasService {
+
+	protected CampaignService(OasApiService apiService) {
+		super(apiService);
+	}
+
+	private final XmlRequestGenerator readCampaignRequestGenerator = new XmlRequestGenerator("read-campaign");
+	private final XmlRequestGenerator addCampaignRequestGenerator = new XmlRequestGenerator("add-campaign");
+	private final XmlRequestGenerator updateCampaignRequestGenerator = new XmlRequestGenerator("update-campaign");
+
+	final public Campaign getCampaignById(final String id) {
```

Comment:
Yes I think its appropriate to rename the methods to getById,add,update,getAll etc. I have done the changes in all the services